src/muse/demand_share.py

```diff
@@ -63,6 +63,17 @@ def demand_share(
 from muse.agents import AbstractAgent
 from muse.registration import registrator
 
+
+class InconsistencyInDemandShare(Exception):
+
+    msg = """A retrofit agent has been found in a 'New agents'-only demand share
+function. Make sure you remove all the retro agents from the Agents input files or use a
+demand share method that can handle both new and retro agents."""
+
+    def __str__(self):
+        return self.msg
+
+
```

Comment:
If we only remove the "retrofit" agents from Agents.csv, we get this error:

 File "c:\users\sg2410\desktop\github\muse_os\src\muse\demand_share.py", line 463, in _inner_split
    total = sum(shares.values()).sum("asset")  # type: ignore
AttributeError: 'int' object has no attribute 'sum'

We may add this to the error lists with a message like:

"This error refers to an agent with no assets. To fix this error, check the capacity assigment to the agents. ONe possibility is that you have decided not to use "Retrofit" agents, as such you may have already removed them from the agent definition file and the file of technodata, the system TOML file should change the demand_share to "standard_demand" function in each subsector section for each of the selected sectors to model.